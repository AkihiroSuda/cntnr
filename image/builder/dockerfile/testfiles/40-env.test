FROM alpine:3.7

ENV VAR1 myval
ENV VAR1 myval1
ENV VARX=myvalx
ENV VARA=80 VARB=443
ENV VARL="long value"


# env substitution

ENV SVAR1=$VAR1
ENV SVAR2=${VAR1}
ENV SVAR3=\$VAR1
ENV SVAR4=\${VAR1}

# TODO: test ADD substitution

COPY $VAR1 /dest
COPY /src $VAR1
COPY --chown=$VAR1 /src /dest

EXPOSE $VARA $VARB

LABEL $VARX=$VARL

STOPSIGNAL $VARA

USER $SVAR1

VOLUME $SVAR1 $VARX

WORKDIR $VARX


# require no substitution

RUN echo $VAR1