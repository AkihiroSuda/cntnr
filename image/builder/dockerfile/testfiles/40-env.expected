FROM alpine:3.7

ENV "VAR1"="myval"
ENV "VAR1"="myval1"
ENV "VARX"="myvalx"
ENV "VARA"="80" "VARB"="443"
ENV "VARL"="long value"


ENV "SVAR1"="myval1"
ENV "SVAR2"="myval1"
ENV "SVAR3"="$VAR1"
ENV "SVAR4"="${VAR1}"

COPY "./ctx" "myval1" "/dest" nil
COPY "./ctx" "/src" "myval1" nil
COPY "./ctx" "/src" "/dest" myval1

EXPOSE 80 443

LABEL "myvalx"="long value"

STOPSIGNAL 80

USER myval1

VOLUME "myval1" "myvalx"

WORKDIR myvalx


RUN "/bin/sh" "-c" "echo $VAR1"